docs/DeveloperGuide.md

```diff
@@ -1,11 +1,262 @@
+
+### 1.4 [Proposed] Search reservation
+#### 1.4.1. Propose Implementation
+
+In the restaurant daily report, users can search against the reservation category by supplying either a reservation number or a date.
+
+The feature implements the following operations:
+* `SearchReservationCommand#parseInput()` - parse the user input for either reservation number or a date.
+* `SearchReservationCommand#execute()` - search along the reservation list for the target reservation(s).
+
+The following class diagram shows the structures relevant to the "search reservation" feature:
+<p align="center">
+    <img src="ClassDiagramforSearchReservation.png">
+</p>
+
+Given below is an example usage scenario and how the search mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. An empty `reservation` will be initialized.
+
+Step 2. The user executes `add reservation; p/Peter; d/2020-03-12 12:00; n/3; c/98955555;` command to add a reservation into the `reservation` list.  
+Further, the user may add more reservations into the current `reservation` list.  
+Suppose the user executes `add reservation; p/Mary; d/2020-03-11 12:00; n/8; c/99998888;`  
+and `add reservation; p/Lisa; m/no spicy food please; d/2020-03-12 12:00; n/3; c/98889999;` as well.
+
+Step 3. The user can now search against the current `reservation` list to see if an reservation is stored in the `reservation` list.  
+If the user executes `search reservation; r/0;`, the following result will be displayed in the image.
+
+<p align="center">
+    <img src="SearchByIndex.png">
+</p>
+
+If the user executes `search reservation; d/2020-03-12;`, the following result will be displayed in the image.
+
+<p align="center">
+    <img src="SearchByDate.png">
+</p>
+
+If the user executes `search reservation; r/0; d/2020-03-13;`, the following result will be displayed in the image.
+
+<p align="center">
+    <img src="SearchByIndexnDate.png">
+</p>
+
+The following sequence diagram shows the relevant interactions behind `search reservation`:
+
+<p align="center">
+    <img src="SequenceDiagramforSearchReservation.png">
+</p>
+
+#### 1.4.2 Design Considerations
+##### Aspect: How search reservation executes
+* **Alternative 1 (current choice)**: Access reservation by `ReservationList#getReservation(reservationNumber)` if reservation number is provided,
+or do linear search along the reservation list for reservations with date matched.
+  + Pros: Easy to implement and understand.
+  + Cons: May have performance issues in terms of time usage when searching along the reservation list.
+
+* **Alternative 2**: Create a `HashMap` with date as the key and all Reservation objects in that date as the value.
+Once a new Reservation object is created, it will be added into the `HashMap` according to its date.
+  + Pros: Easier for searching in terms of date.
+  + Cons: Extra memory space is needed.
+
+**The reason to make the current choice:**  
```

Comment:
This seems to be inconsistent with the other commands and perhaps not necessary if the reason is relatively clear given the pros and cons listed for the current choice and the alternative.

Reply:
Thanks for your suggestion!